js/web/lib/wasm/session-handler.ts

```diff
@@ -119,128 +119,165 @@ const getLoggingLevel = (loggingLevel: 'verbose'|'info'|'warning'|'error'|'fatal
   }
 };
 
-const setSessionOptions = (options?: InferenceSession.SessionOptions): [number, number[]] => {
-  const wasm = getInstance();
-  const sessionOptionsHandle = wasm._OrtCreateSessionOptions();
-  const allocs: number[] = [];
-
-  if (sessionOptionsHandle === 0) {
-    throw new Error('Can\'t create session options');
+const getGraphOptimzationLevel = (graphOptimizationLevel: 'disabled'|'basic'|'extended'|'all'): number => {
+  switch (graphOptimizationLevel) {
+    case 'disabled':
+      return 0;
+    case 'basic':
+      return 1;
+    case 'extended':
+      return 2;
+    case 'all':
+      return 99;
+    default:
+      throw new Error(`unsupported graph optimization level: ${graphOptimizationLevel}`);
   }
+};
 
-  if (options === undefined) {
-    return [sessionOptionsHandle, allocs];
+const getExecutionMode = (executionMode: 'sequential'|'parallel'): number => {
+  switch (executionMode) {
+    case 'sequential':
+      return 0;
+    case 'parallel':
+      return 1;
+    default:
+      throw new Error(`unsupported execution mode: ${executionMode}`);
   }
+};
 
-  let errorCode = 0;
-
-  if (options.graphOptimizationLevel !== undefined) {
-    switch (options.graphOptimizationLevel) {
-      case 'disabled':
-        errorCode = wasm._OrtSetSessionGraphOptimizationLevel(sessionOptionsHandle, 0);
-        break;
-      case 'basic':
-        errorCode = wasm._OrtSetSessionGraphOptimizationLevel(sessionOptionsHandle, 1);
-        break;
-      case 'extended':
-        errorCode = wasm._OrtSetSessionGraphOptimizationLevel(sessionOptionsHandle, 2);
-        break;
-      case 'all':
-        errorCode = wasm._OrtSetSessionGraphOptimizationLevel(sessionOptionsHandle, 99);
-        break;
-      default:
-        throw new Error(`unsupported graph optimization level: ${options.graphOptimizationLevel}`);
-    }
-    if (errorCode !== 0) {
-      throw new Error(`Can't set a graph optimization level as a session option. error code = ${errorCode}`);
-    }
-  }
+interface ExtraOptionsHandler {
+  (name: string, value: string): void;
+}
 
-  if (options.enableCpuMemArena !== undefined) {
-    if (options.enableCpuMemArena) {
-      errorCode = wasm._OrtEnableCpuMemArena(sessionOptionsHandle);
+const iterateExtraOptions = (options: Record<string, unknown>, prefix: string, handler: ExtraOptionsHandler) => {
+  Object.entries(options).forEach(([key, value]) => {
+    const name = (prefix) ? prefix + '.' + key : key;
+    if (typeof value === 'object') {
+      iterateExtraOptions(value as Record<string, unknown>, name, handler);
+    } else if (['string', 'number', 'boolean'].indexOf(typeof value) !== -1) {
```

Comment:
do we allow users to specify non-string value for each config?

if we do, we should serialize boolean to "1" or "0" because "true" and "false" don't work for those options.